app/models/teacher_training_adviser/steps/stage_of_degree.rb

```diff
@@ -18,6 +19,13 @@ class StageOfDegree < GITWizard::Step
 
     include FunnelTitle
 
+    def save
+      if (creation_channel_source_id.present? || creation_channel_activity_id.present? || channel_id.nil?) && (tta? || !creation_channel_service_id.in?(creation_channel_service_ids))
```

Comment:
sorry to be annoying but could this complex conditional be extracted to a method with a name reflecting what it does?

Reply:
good shout, will fix